std/protocols/lookup_via_bus.asm

```diff
@@ -2,28 +2,24 @@ use std::array::len;
 use std::check::assert;
 use std::protocols::bus::bus_send;
 use std::protocols::bus::bus_receive;
-use std::protocols::bus::compute_next_z_send;
-use std::protocols::bus::compute_next_z_receive;
 use std::protocols::lookup::unpack_lookup_constraint;
 use std::math::fp2::Fp2;
+use std::constraints::to_phantom_lookup;
 
```

Comment:
Why not keep the simple `lookup` function that just calls send and receive?

Reply:
Because we'd want to do the receive in the callee machine and the send in the caller machine. Otherwise, the added columns would be in the wrong namespace.

Reply:
But namespaces will not matter later on anyway...

Reply:
How do they matter? Only wrt degree, or is there something else?

Reply:
We split by namespace in our [`split_pil`](https://github.com/powdr-labs/powdr/blob/2219cf3aa5559e426977bddc65de6063bbc58767/backend-utils/src/lib.rs#L29) function, used by the composite backend and Plonky3 backend to split the columns (and constraints) into machines, so we can prove them even when they a have different number of rows. I think this will go away once we have proof objects.

If we had one `lookup` function that was called in the machine linking to the callee (just like our current ASM linker adds the native lookup constraint to the caller), the multiplicity column would end up in the wrong namespace, which would lead to an error in the `split_pil` function, because it doesn't expect constraints to reference columns in multiple namespaces.